src/endpoint/s3/s3_utils.js

```diff
@@ -322,6 +324,18 @@ function set_response_object_md(res, object_md) {
     }
 }
 
+/**
+ * @param {nb.S3Response} res 
+ * @param {Array<string>} [supported_storage_classes]
+ */
+function set_response_supported_storage_classes(res, supported_storage_classes = []) {
+    if (!config.DENY_UPLOAD_TO_STORAGE_CLASS_STANDARD) {
+        supported_storage_classes.push(STORAGE_CLASS_STANDARD);
```

Comment:
so now the storage classes returned by the bucket always do not include STANDARD, which just becomes confusing. I didn't mean that. I only meant that if the bucketspace does not return anything, that we set the default based on the config too, but separating the responsibility like this is too weird. So I think we should just do this instead - have the bucketspace return the list, and if it doesn't return anything, just not set the header.